src/dbnode/server/server.go

```diff
@@ -564,9 +565,27 @@ func Run(runOpts RunOptions) {
 
 	if cfg.DebugListenAddress != "" {
 		go func() {
-			if err := http.ListenAndServe(cfg.DebugListenAddress, nil); err != nil {
+			mux := http.DefaultServeMux
+			cpuProfileDuration := 1 * time.Second
+			debugWriter, err := xdebug.NewZipWriterWithDefaultSources(
```

Comment:
Can you actually initialize this somewhere outside of this Goroutine? Fairly early on in `server.go` would be ideal. The reason I ask that is we're going to want to register new sources soon (one to dump the YAML file, another to dump the namespace config) so we'll want it available more broadly than just in this async routine

Reply:
Makes sense. I moved it further up, right after initialisation of `iopts`